src/features/survey-forms/[id]/survey-form-table.tsx

```diff
@@ -1,338 +1,257 @@
+        },
+      ])
+    } else {
+      setTotalAmount((prev) => prev - choiceAmount)
+      setSelectedSurveyAnswerIds((prev) => prev.filter((id) => id !== choiceId))
+      setSurveyAnswers(
+        surveyAnswers.filter((answer) => answer.survey_template_answer_id !== choiceId)
+      )
+    }
+  }
+
+  const handleSubmit = async () => {
+    try {
+      if (id !== undefined) {
+        const result = await appDispatch(
+          submitSurveyAnswers({
+            survey_administration_id: parseInt(id),
+            survey_answers: surveyAnswers,
+          })
+        )
+        if (result.type === "user/submitSurveyAnswers/rejected") {
+          appDispatch(
+            setAlert({
+              description: result.payload,
+              variant: "destructive",
+            })
+          )
+        }
+        if (result.type === "user/submitSurveyAnswers/fulfilled") {
+          appDispatch(
+            setAlert({
+              description: "Successfully submitted survey answers",
+              variant: "success",
+            })
+          )
+          appDispatch(updateSurveyResultStatus(SurveyResultStatus.Completed))
+        }
+      }
+    } catch (error) {}
+  }
 
   return (
     <div>
       <div>
-        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Voluptates, natus odio! Eaque
-        necessitatibus molestiae, fugiat nam labore modi sed saepe nemo consequuntur voluptatum
-        maxime soluta eum tenetur? Adipisci nostrum deserunt quod cum amet ipsum.
-      </div>
-      <div className='mt-10'>
-        {Questions.map((question) => (
-          <div key={question.id} className='mb-10'>
-            <p className='mb-1 font-bold flex items-center gap-5'>
-              {question.id}
-              {"."} {question.question}
-              <Badge variant='darkPurple' size='medium'>
-                {0}
-              </Badge>
-            </p>
-            <div className='flex flex-row gap-4 overflow-auto whitespace-nowrap'>
-              {categories.map((category, index) => (
-                <div key={index}>
-                  <div className='mb-4'>
-                    <Button fullWidth variant={"primaryOutline"} size='small' fullHeight>
-                      {category}
-                    </Button>
-                  </div>
+        {loading === Loading.Pending && (
+          <div className='text-center'>
+            <Spinner />
+          </div>
+        )}
+        {loading === Loading.Fulfilled && user_survey_questions !== null && (
+          <>
+            {user_survey_questions.map((question) => (
+              <div key={question.id} className='mb-10'>
+                <p className='mb-1 font-bold flex items-center gap-5'>
+                  {question.id}
+                  {"."} {question.question_text}
+                  {question.is_required === true && <span className='text-red-500'>*</span>}
+                  <Badge variant='darkPurple' size='medium'>
```

Comment:
can we pls add label for Total Amount here. also can place this under question and check if it looks okay else can retain on current position